### ISSUE TITLE
Confusion about reward assignment in PSFT

### ISSUE THREAD

**Issue by NeoDH** [zwhong714/PSFT#8]
Hello, thank you for open-sourcing this work!

I am new to Verl and was going through the PSFT recipe. Since PSFT is a variant of SFT, the reward for each token is expected to be initialized as 1. However, I found that the reward token_level_scores is instead assigned from responses:

https://github.com/zwhong714/PSFT/blob/930e23980a723ecef5af138e6e32aa3798b1fd64/verl/recipe/psft/psft_ray_trainer.py#L174

If I understand correctly, new_batch.batch["responses"] contains the generated token ids ([bs, seq_len]), as seen in:

https://github.com/zwhong714/PSFT/blob/930e23980a723ecef5af138e6e32aa3798b1fd64/verl/verl/workers/rollout/vllm_rollout/vllm_rollout_spmd.py#L373

Did I miss some step in the reward function processing?

**Comment by zwhong714** [comment]
On line 365, the statement `outputs = non_tensor_batch['demonstration']` assigns data that is not generated by the model.

**Comment by NeoDH** [comment]
Yes, I have noticed it. Then, does batch['responses'] denote the token IDs of the demonstration?

I am confused about why `new_batch.batch["token_level_scores"]` is assigned as 
`torch.ones_like(new_batch.batch["responses"], dtype=torch.float32) * new_batch.batch["responses"]` 
instead of simply `torch.ones_like(new_batch.batch["responses"], dtype=torch.float32)`.

**Comment by zwhong714** [comment]
Thank you for pointing out this mistake.
Originally, we implemented the token reward assignment within the advantage calculation function.

To make the PSFT modification more explicit, we extracted it into psft_ray_trainer.py. Specifically, what we want to implemented is as:

`torch.ones_like(new_batch.batch["responses"], dtype=torch.float32) * new_batch.batch["response_mask"]`

We will revise this code accordingly.

Thank you again for your helpful feedback!

**Comment by NeoDH** [comment]
Thanks for your explanation! It makes sense.

One more thing I want to check: when I reproduce it, is the only change for the reward that new_batch.batch["response_mask"] replaces new_batch.batch["responses"], or are there any corresponding modifications in other places as well?

**Comment by zwhong714** [comment]
Wait, `token_level_rewards` will not be used going forward.

The actual functionality is implemented(https://github.com/zwhong714/PSFT/blob/930e23980a723ecef5af138e6e32aa3798b1fd64/verl/verl/trainer/ppo/ray_trainer.py#L274-L276)

**Comment by NeoDH** [comment]
Yes, the final computation of advantages does not involve token_level_scores or token_level_rewards. The code on line 174 appears to be misspelled and just a dummy process. I close this issue. Thanks again.
